DataLayer/FastaReader.h

```diff
@@ -76,32 +88,38 @@ class FastaReader {
 		/** Read a single line. */
 		std::istream& getline(std::string& s)
 		{
-			if (std::getline(m_in, s)) {
+			if (std::getline(*m_in, s)) {
 				chomp(s, '\r');
 				m_line++;
 			}
-			return m_in;
+			return *m_in;
 		}
 
 		/** Ignore the specified number of lines. */
 		std::istream& ignoreLines(unsigned n)
 		{
 			for (unsigned i = 0; i < n; ++i) {
-				if (m_in.ignore(
+				if (m_in->ignore(
 						std::numeric_limits<std::streamsize>::max(),
 						'\n'))
 					m_line++;
 			}
-			return m_in;
+			return *m_in;
 		}
 
 		std::ostream& die();
 		bool isChaste(const std::string& s, const std::string& line);
 		void checkSeqQual(const std::string& s, const std::string& q);
 
 		const char* m_path;
-		std::ifstream m_fin;
-		std::istream& m_in;
+		std::ifstream* m_fin;
+		std::istream* m_in;
```

Comment:
Please leave both of these as objects rather than pointers. It's okay if the file is never opened when using `kseq`.